packages/eslint-plugin/rules/__tests__/data-no-store-string-literals.js

```diff
@@ -57,6 +57,85 @@ const invalid = [
 	`import { controls } from '@wordpress/data'; controls.dispatch( 'core' );`,
 	`import { controls } from '@wordpress/data'; controls.resolveSelect( 'core' );`,
 	`import { controls as controlsAlias } from '@wordpress/data'; controlsAlias.resolveSelect( 'core' );`,
+
+	// Direct function calls suggestions
+	{
+		// Replace core with coreStore and import coreStore
+		code: `import { select } from '@wordpress/data'; select( 'core' );`,
+		errors: [
+			{
+				suggestions: [
+					{
+						desc:
+							'Replace literal with store definition. Import store if neccessary.',
+						output: `import { select } from '@wordpress/data';\nimport { store as coreStore } from '@wordpress/core-data'; select( coreStore );`,
+					},
+				],
+			},
+		],
+	},
+	{
+		// Replace core with coreStore. A @wordpress/core-data already exists, so it should append the import to that one.
+		code: `import { select } from '@wordpress/data'; import { something } from '@wordpress/core-data'; select( 'core' );`,
+		errors: [
+			{
+				suggestions: [
+					{
+						desc:
+							'Replace literal with store definition. Import store if neccessary.',
+						output: `import { select } from '@wordpress/data'; import { something,store as coreStore } from '@wordpress/core-data'; select( coreStore );`,
+					},
+				],
+			},
+		],
+	},
+	{
+		// Replace core with coreStore. A @wordpress/core-data already exists, so it should append the import to that one.
+		// This time there is a comma after the import.
+		code: `import { select } from '@wordpress/data'; import { something, } from '@wordpress/core-data'; select( 'core' );`,
+		errors: [
+			{
+				suggestions: [
+					{
+						desc:
+							'Replace literal with store definition. Import store if neccessary.',
+						output: `import { select } from '@wordpress/data'; import { something,store as coreStore, } from '@wordpress/core-data'; select( coreStore );`,
```

Comment:
The output isn't following the coding styles but I guess it's easy to fix with Prettier.

Reply:
I'd rather rely on multipass fixes 😄 Basically I apply the fix and save it immediately, thus it fixes all the auto-fixable problems. If you think it's worth fixing this I'm happy to do it though. Let me know

Source: https://eslint.org/docs/developer-guide/working-with-rules
> Note: Suggestions will be applied as a stand-alone change, without triggering multipass fixes. Each suggestion should focus on a singular change in the code and should not try to conform to user defined styles. For example, if a suggestion is adding a new statement into the codebase, it should not try to match correct indentation, or confirm to user preferences on presence/absence of semicolons. All of those things can be corrected by multipass autofix when the user triggers it.

Reply:
Yes, it makes perfect sense. I considered it when mentioning Prettier, but now I see this recommendation and I totally get the idea. Even when you add spaces here, something else might add another variable to the import or reorder them and finally Prettier will do its own changes regardless.